### PR TITLE
Bug/vspc 207

### PR DESCRIPTION
**Pull request by swetalina-prusty25**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [ ] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

When going to "Images" on the staff side, right now it shows a table of all images. There should be a search to filter images. The search should search filename, name, and description for the given search term.

  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

**Comment by swetalina-prusty25** [comment]
Removed all indentation error

File: vspace/src/main/java/edu/asu/diging/vspace/core/data/ImageRepository.java [hunk 1]
```diff
@@ -14,15 +13,19 @@
...
+    Page < VSImage > findByFilenameLikeOrNameLikeOrDescriptionLike(Pageable pageable, String filename, String name, String description);
 
-    List<VSImage> findByFilenameLikeOrNameLike(String filename, String name);
+    List < VSImage > findByFilenameLike(String filename);
 
-    List<VSImage> findByFilenameLike(String filename);
+    Page < VSImage > findByCategories(Pageable pageable, ImageCategory category);
 
-    Page<VSImage> findByCategories(Pageable pageable, ImageCategory category);
+    List < IVSImage > findByCategories(ImageCategory category);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/data/ImageRepository.java:28]: the shouldn't be blanks after/before `<` and `>`

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 2]
```diff
@@ -250,4 +283,43 @@ public void removeCategory(IVSImage image, ImageCategory category) {
...
+        pageNo = validatePageNumber(pageNo, images);
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize);
+        String likeSearchTerm = "%" + searchTerm + "%";
+        Page < VSImage > result = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(sortByRequestedField,
+            likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        List < IVSImage > imageResults = new ArrayList < > ();
+        if (result != null) {
+            result.getContent().forEach(i - > imageResults.add(i));
+        }
+        return imageResults;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:323]: I'm very confused by what this method does. Why does it get a list of images and then also search the repo? It should jus search and return the request page of search results.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImageSearchApiController.java [hunk 3]
```diff
@@ -18,31 +15,29 @@
...
         List<IVSImage> images = null;
         if (searchTerm != null && !searchTerm.trim().isEmpty()) {
-            images = imageService.findByFilenameOrNameContains(searchTerm);
-        } else {
-            images = imageService.getImages(1);
-        }
-        
+            images = imageService.findByFilenameOrNameContains(searchTerm);}
+        else {
+            images = imageService.getImages(1);}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImageSearchApiController.java:28]: curly bracket needs fixing

Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImageSearchApiController.java:28]: code factor issue will probably go away after. if not this also needs fixing!

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            if (spaces != null && spaces.size() > 0) {
+                imageToSpaces.put(image.getId(), spaces);
+            }
+	}
+        model.addAttribute("imageToSpaces", imageToSpaces);
+        model.addAttribute("images", imageResults);
+        model.addAttribute("searchTerm", searchTerm);
+        return "staff/images/searchImages";
+    }
+}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:70]: indentation

File: vspace/src/main/webapp/WEB-INF/views/staff/images/imagelist.html [hunk 5]
```diff
@@ -1,204 +1,243 @@
...
+            $(document).on('keydown', function(e) { 
+            
+                if((e.key  === 'Enter' || e.keyCode === 13) && $("#searchImageField").val().trim()) {
+               	 
+                    $("#searchImageButton").trigger('click');
+                }
+            }) 
+         </script>
+      </div>
+   </body>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/images/imagelist.html:242]: only the parts that are actually changed should be changed. The way it is right now, I can't see what you have actually change in this file.

**Review by jdamerow — CHANGES_REQUESTED** [review]
> I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)

You checked this, but you did it anyways.

>  I have explained any part of my code/implementation decisions that is not be self-explanatory

why didn't you check this box?

File: vspace/src/main/java/edu/asu/diging/vspace/core/data/ImageRepository.java (hunk 1 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/data/ImageRepository.java:28]: Removed blanks from <>.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java (hunk 2 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:323]: Requirement says "The search should search filename, name, and description for the given search term"
so in this function i am sending IVS images to validate the page number. and searching in repo to get the list of images for that particular page based on search conditions.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java (hunk 4 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:70]: Resolved Indentation issues for this file.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java (hunk 2 above)
Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:323]: but why does it need a list off images for that?

Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:323]: I am using an existing function validatePageNumber() which is present in Imageservice.java file, that function has list of images as parameter, so sending the list of images to validate the function which will return page no, then based on the page number requesting images in the specific sorted order.

File: vspace/src/main/webapp/WEB-INF/views/staff/images/imagelist.html (hunk 5 above)
Comment (swetalina-prusty25) [reply vspace/src/main/webapp/WEB-INF/views/staff/images/imagelist.html:242]: Took the version file from develop and added my code for this ticket again.

**Pull request closed without merging**